### PR TITLE
[mezmoexporter] Test refactoring

### PR DESCRIPTION
**Pull request by jsumners**
Some light refactoring in the `mezmoexporter` test suite to facilitate easier testing in the future.

### CODE REVIEW

File: exporter/mezmoexporter/exporter_test.go [hunk 1]
```diff
@@ -96,54 +96,88 @@ func createSizedPayloadLogData(payloadSize int) plog.Logs {
...
 }
 
-func TestLogsExporter(t *testing.T) {
-	// Spin up a HTTP server to receive the test exporters...
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
-		assert.Equal(t, "mezmo-otel-exporter/"+buildInfo.Version, r.Header.Get("User-Agent"))
+type testServer struct {
+	Instance *httptest.Server
+	Url      string
```
Comment (dashpole) [inline exporter/mezmoexporter/exporter_test.go:101]:
```suggestion
	instance *httptest.Server
	url      string
```
nit: keep these unexported

File: exporter/mezmoexporter/exporter_test.go [hunk 2]
```diff
@@ -96,54 +96,88 @@ func createSizedPayloadLogData(payloadSize int) plog.Logs {
...
+		assertionsCallback: func(req *http.Request, body MezmoLogBody) {
+			assert.Equal(t, "application/json", req.Header.Get("Content-Type"))
+			assert.Equal(t, "mezmo-otel-exporter/"+buildInfo.Version, req.Header.Get("User-Agent"))
+		},
+	}
+	server := createHttpServer(&httpServerParams)
+	defer server.Instance.Close()
+
+	config := &Config{
+		IngestURL: server.Url,
```
Comment (dashpole) [inline exporter/mezmoexporter/exporter_test.go:162]:
```suggestion
		IngestURL: server.url,
```

**Comment by jsumners** [comment]
I think the "skip changelog" label is appropriate here. But I will update the changelog if it is desired.

File: exporter/mezmoexporter/exporter_test.go [hunk 3]
```diff
@@ -96,54 +96,88 @@ func createSizedPayloadLogData(payloadSize int) plog.Logs {
...
+func TestLogsExporter(t *testing.T) {
+	httpServerParams := testServerParams{
+		t: t,
+		assertionsCallback: func(req *http.Request, body MezmoLogBody) {
+			assert.Equal(t, "application/json", req.Header.Get("Content-Type"))
+			assert.Equal(t, "mezmo-otel-exporter/"+buildInfo.Version, req.Header.Get("User-Agent"))
+		},
+	}
+	server := createHttpServer(&httpServerParams)
+	defer server.Instance.Close()
```
Comment (dashpole) [inline exporter/mezmoexporter/exporter_test.go:159]:
```suggestion
	defer server.instance.Close()
```

File: exporter/mezmoexporter/exporter_test.go (hunk 1 above)
Comment (jsumners) [reply exporter/mezmoexporter/exporter_test.go]: Corrected.

**Review by dashpole — APPROVED** [review]

**Comment by jsumners** [comment]
Rebased on upstream `main` to resolve CI failures.

**Review by pmm-sumo — APPROVED** [review]

**Pull request merged**